key.ui/src/main/java/de/uka/ilkd/key/gui/actions/HeatmapToggleAction.java

```diff
@@ -67,5 +69,11 @@ public void actionPerformed(ActionEvent e) {
         // this updates the heatmap highlights
         mainWindow.getCurrentGoalView().getHighlighter().removeAllHighlights();
         mainWindow.getCurrentGoalView().printSequent();
+        Component component = mainWindow.getMainFrame().getContent();
```

Comment:
This should now always update twice, right? See my suggestion ...

Reply:
Don't know. If it works with your suggestion, then ... yes. I was not sure. Thanks for finding out.